### PR TITLE
Added changes for new note UI and addressed review comments from prev PRs #3

### PR DESCRIPTION
**Pull request by nivetha-yadhavan**
https://user-images.githubusercontent.com/30879931/182978572-f3dd3517-8339-4599-afce-35feafb41c11.mp4

### CODE REVIEW

File: app/javascript/src/components/Dashboard/Notes/Pane/Form.jsx [hunk 1]
```diff
@@ -39,15 +32,43 @@ export default function NoteForm({ onClose, refetch, note, isEdit }) {
...
               className="w-full flex-grow-0"
               required
             />
             <Textarea
               label="Description"
               name="description"
+              placeholder="Enter note description"
+              className="w-full flex-grow-0"
+              rows={1}
+              required
```
Comment (yedhink) [inline app/javascript/src/components/Dashboard/Notes/Pane/Form.jsx]: Implicitly `true` attributes should be kept at the top of the component's prop list.

Refer: https://academy.bigbinary.com/learn-react/pages/miscellaneous-tips#:~:text=Also%20note%20that,Next

File: app/javascript/src/components/Dashboard/Notes/constants.js [hunk 2]
```diff
@@ -4,11 +4,18 @@ import * as yup from "yup";
...
+  tags: "",
 };
 
 export const NOTES_FORM_VALIDATION_SCHEMA = yup.object().shape({
   title: yup.string().required("Title is required"),
   description: yup.string().required("Description is required"),
+  assignedContact: yup
+    .object()
+    .required("Assigned Contacts is required")
+    .nullable(),
```
Comment (yedhink) [inline app/javascript/src/components/Dashboard/Notes/constants.js]: I am kinda confused here. What does it mean for an entity to be `required` as well as be `nullable`? Isn't it counter intuitive?

PS: If there's a valid business aspect for adding this logic, then we can keep it. I am just trying to understand the same.

File: app/javascript/src/components/Dashboard/Notes/constants.js [hunk 3]
```diff
@@ -4,11 +4,18 @@ import * as yup from "yup";
...
+  tags: yup.object().required("Tags is required").nullable(),
```
Comment (yedhink) [inline app/javascript/src/components/Dashboard/Notes/constants.js]: Same as above

File: app/javascript/src/components/Dashboard/Notes/index.jsx [hunk 4]
```diff
@@ -6,24 +6,24 @@ import { Container, Header } from "neetoui/layouts";
...
 import NewNotePane from "./Pane/Create";
 
 const Notes = () => {
   const [showNewNotePane, setShowNewNotePane] = useState(false);
   const [showDeleteAlert, setShowDeleteAlert] = useState(false);
   const [searchTerm, setSearchTerm] = useState("");
   const [showMenu, setShowMenu] = useState(false);
-  const [notesList, setNotes] = useState(notes);
-  const [selectedNoteId, setSelectedNoteId] = useState();
+  const [notesList, setNotesList] = useState(NOTES);
```
Comment (yedhink) [inline app/javascript/src/components/Dashboard/Notes/index.jsx]: You don't have to suffix with the keyword `List` in order to denote that the resource is a collection.

Using plural variable name suggests the same:
```suggestion
  const [notes, setNotes] = useState(NOTES);
```

**Review by yedhink — CHANGES_REQUESTED** [review]
@nivetha-yadhavan _a Few comments added. Once you fix them, please go ahead and merge this PR.

**Comment by yedhink** [comment]
@nivetha-yadhavan I am not sure why you've assigned this PR to me now. Had you fixed the requested review changes and pushed those changes? If the assignment was done accidentally, then please remove my assignment. Thanks.

image

File: app/javascript/src/components/Dashboard/Notes/constants.js (hunk 2 above)
Comment (nivetha-yadhavan) [reply app/javascript/src/components/Dashboard/Notes/constants.js]: That's my bad. I was getting a null error and was trying out a few things. Turned out the issue was with the way I had declared the objects for these fields and I missed to remove the nullable here.

File: app/javascript/src/components/Dashboard/Notes/index.jsx [hunk 5]
```diff
@@ -33,7 +33,7 @@ const Notes = () => {
 
   return (
     <>
-      <NotesMenu showMenu={showMenu}></NotesMenu>
+      <Menu showMenu={showMenu}></Menu>
```
Comment (yedhink) [inline app/javascript/src/components/Dashboard/Notes/index.jsx:36]: Use inline closing in cases where there are no children for the component.
```suggestion
<Menu showMenu={showMenu} />
```

**Review by yedhink — COMMENTED** [review]
@nivetha-yadhavan One more comment added.

File: app/javascript/src/components/Dashboard/Notes/index.jsx (hunk 5 above)
Comment (nivetha-yadhavan) [reply app/javascript/src/components/Dashboard/Notes/index.jsx:36]: Fixed in https://github.com/nivetha-yadhavan/wheel/pull/10/

**Pull request merged**